src/routes/authRouter.js

```diff
@@ -7,23 +7,134 @@ import UserMiddleware from '../middlewares/UserMiddleware';
 
 
 const authRouter = express.Router();
-
+/**
+   * @swagger
+   * /api/v1/auth/signup:
+   *   post:
+   *     description: Signup to the application
+   *     produces:
+   *       - application/json
+   *     parameters:
+   *       - name: email
+   *         description: User's email.
+   *         in: formData
+   *         required: true
+   *         type: string
+   *       - name: password
+   *         description: User's password.
+   *         in: formData
+   *         required: true
+   *         type: string
+   *       - name: userType
+   *         description: User's type.
+   *         in: formData
+   *         required: true
+   *         type: string
+   *       - name: signUpType
+   *         description: User's type.
+   *         in: formData
+   *         required: true
+   *         type: string
+   *     responses:
+   *       400:
+   *         description: There was a problem sending
+   *         schema:
+   *           type: object
+   *       200:
+   *         description: User created successfully and verification link sent to your Email
+   *         schema:
+   *           type: object
+   */
 authRouter
   .post('/signup',
     UserValidator.UserSignUpValidator,
     emailCheck,
     UserController.signUp);
+/**
+   * @swagger
```

Comment:
Any reason for using swagger JSDOCs against Swagger.json where the documentation is kept in a different file. I feel using swagger.json file and having all the documentation there is cleaner because adding documentation to the route file will make it unavoidably long and I don't think we want that. Anyone going through the route file will have to read the doc of each of them.

Reply:
Ok.